lib/rules/display-name.js

```diff
@@ -90,13 +90,13 @@ module.exports = {
       const namedClass = (
         (node.type === 'ClassDeclaration' || node.type === 'ClassExpression') &&
         node.id &&
-        node.id.name
+        !!node.id.name
```

Comment:
This and the below addition of `!!` aren't strictly needed but I spotted every other variable in this block returns a boolean so I figured these should too